### ISSUE TITLE
Problems specifying "backingfile " in snp_readBGEN

### ISSUE THREAD

**Issue by alesss78** [privefl/bigsnpr#34]
I am trying to read a BGEN file using `bigsnpr::snp_readBGEN()`.
I am using the following code in a google virtual machine with 100 GB ram and 16 cores.

```
system.time(
  rds <- bigsnpr::snp_readBGEN(
    bgenfiles = glue::glue("/data/bgen/ukb_imp_chr{chr}_v3.bgen", chr = 1:22),
    list_snp_id = list_snp_id,
    backingfile = "data/try_1",
    ind_row = ind.indiv[sub],
    bgi_dir = "/data/bgen",
    ncores = 12
  )
)
```

However, I get an error that is related to the specification of the backingfile.
In particular, the error I encounter the following error:

```
Error in getXPtrFBM(.self$bk, .self$nrow, .self$ncol, .self$type) : 
  Error when mapping file:
  No such file or directory.
In addition: Warning message:
In normalizePath(bkfile) :
  path[1]="/data/try_1.bk": No such file or directory
Timing stopped at: 0.032 0 0.033
```

In my understading, the backingfile  is created anew during snp_readBGEN execution.
However, I get the error that specifies that "No such file or directory" exists.

I tried using an alternative line to specify the backingfile :
`backingfile = tempfile(pattern = "lasso_try_1")`
It works. However, the temporary file is created in the temporary folder of R and the file is too big. I would need to specify a different server location outside of the temporary folder of R.

**Comment by privefl** [comment]
Thanks for reporting.
Do you have the same error using `bigstatsr::FBM(10, 10, backingfile = "data/try_2")`?

**Comment by alesss78** [comment]
DeaR privefl

Sorry It was a stupid mistake of mine.
I was trying to set a path for the saving of the bk file that was outside of the working directory of R
I changed working directory and now it seems to work

Sorry again for the inconvenience

**Comment by privefl** [comment]
No problem.

**Comment by alesss78** [comment]
Sorry to bother again

I also encountered a different error. When I change working directory and using the following code

system.time(
  rds <- bigsnpr::snp_readBGEN(
    bgenfiles = glue::glue("/data/bgen/ukb_imp_chr{chr}_v3.bgen", chr = 1:22),
    list_snp_id = list_snp_id,
    backingfile = tempfile(pattern = "cad_lasso_try_1"),
    ind_row = ind.indiv[sub],
    bgi_dir = "/data/bgen",
    ncores = 12
  )
)

I get the error : 

Error in unserialize(socklist[[n]]) : error reading from connection
Timing stopped at: 2.216 1.136 341.9

**Comment by privefl** [comment]
Please make sure to select your code and use the "Insert code button (<>)", so that your code and errors are easier to read.

**Comment by alesss78** [comment]
Sorry you are absolutely right

`system.time(
rds <- bigsnpr::snp_readBGEN(
bgenfiles = glue::glue("/data/bgen/ukb_imp_chr{chr}_v3.bgen", chr = 1:22),
list_snp_id = list_snp_id,
backingfile = tempfile(pattern = "cad_lasso_try_1"),
ind_row = ind.indiv[sub],
bgi_dir = "/data/bgen",
ncores = 12
)
)`

**Comment by privefl** [comment]
Any update on this? 
You didn't not seem to have updated your question.

**Comment by alesss78** [comment]
Dear Florian
Thank you for the interest.
The latest error, the one specified by the messagge
"Error in unserialize(socklist[[n]]) : error reading from connection"
has been solved by increasing the memory of the server. I increased RAM from 40 to 80 GB 
and the error stopped happening.
Since the error message was so general, It took a bit for me to realize It could be due to a memory issue.
Thank you again for your help and support.
